### ISSUE TITLE
Goblin Psychopath issue with multiple damage recipients in combat

### ISSUE THREAD

**Issue by Zzooouhh** [magefree/mage#4157]
I'm reposting this here mostly as a self-reminder - Goblin Psychopath's implementation only has its effect redirect a single portion of the damage it deals on a lost flip: i.e. if he is blocked by a single blocker, Psychopath redirects all damage back to me as proper. If there are more blockers or Psychopath has trample, he only redirects damage from the first blocker and everything else gets through. Actually all (non-first/double strike) combat damage should be happening simultaneously, therefore all damage from Psychopath should be redirected the first time he deals it in a turn. The core issues is that all of these damage portions are treated as separate damage events in Xmage.

**Comment by Zerrisx** [comment]
Is this caused by the same root as #4015?

**Comment by awjackson** [comment]
Support was added for "batching" combat damage after the fact (for "dealt damage" triggers), but I believe redirecting/replacing combat damage as a batch is a separate issue.
